backend/authorization/interface.go

```diff
@@ -0,0 +1,10 @@
+package authorization
+
+import "github.com/sensu/sensu-go/types"
+
+// Authorizer determines whether a request is authorized using the RequestInfo
+// passed. It returns true if the request should be authorized, along with any
+// error encountered
+type Authorizer interface {
+	Authorize(reqInfo *types.RequestInfo) (bool, error)
```

Comment:
:+1:

Reply:
I feel like `Authorize ()` being `RequestInfo -> Bool` is a bit strange. It sounds like "authorizing request information". Maybe it should be `Authorize(req http.Request) (bool, error)` or we might want to find a better name for `RequestInfo`...

Reply:
I want to avoid passing `req http.Request` to Authorizers because they are not exclusive to the REST API; we will probably need to implement similar on the agentd side to ensure agents have the right to create events on certain namespaces for example.

I'd like to move the `types.RequestInfo` struct to maybe the `authorization` package and name it something like `Attributes`.

WDYT?

Reply:
`RequestInfo` might be too general for now anyway, as only the Authorization middleware uses it really. And it runs the risk of becoming `BigFatBagOfStuff` as maybe other middlewares piggyback on it in order to not have to define a new type.

I'm happy for you to move it in the `authorization` package. As for its new name... your guess is as good as mine!

`Operation` comes to mind as I feel it's natural to "authorize an operation". But at the same time, `Operation` is also rather general.